stdlib/http/src/test/java/org/ballerinalang/stdlib/services/configuration/ResourceConfigPathTest.java

```diff
@@ -61,14 +61,14 @@ public void testPathParamAndSignatureParamMatch() {
                 .compile("test-src/services/configuration/resource-arg--pathparam-match.bal");
         Diagnostic[] diag = compileResult.getDiagnostics();
         Assert.assertEquals(diag.length, 8);
-        assertResponse(diag[0], INVALID_RESOURCE_PARAMETERS, 10);
-        assertResponse(diag[1], INVALID_RESOURCE_PARAMETERS, 18);
-        assertResponse(diag[2], INVALID_RESOURCE_PARAMETERS, 43);
-        assertResponse(diag[3], "Empty data binding param value", 46);
+        assertResponse(diag[0], INVALID_RESOURCE_PARAMETERS, 14);
+        assertResponse(diag[1], INVALID_RESOURCE_PARAMETERS, 22);
+        assertResponse(diag[2], "Empty data binding param value", 46);
+        assertResponse(diag[3], INVALID_RESOURCE_PARAMETERS, 48);
         assertResponse(diag[4], "Invalid data binding param in the signature : expected 'naMe', but found 'name'", 55);
-        assertResponse(diag[5], INVALID_RESOURCE_PARAMETERS, 61);
-        assertResponse(diag[6], "Invalid data binding param in the signature : expected 'naMe', but found 'name'", 63);
-        assertResponse(diag[7], INVALID_RESOURCE_PARAMETERS, 69);
+        assertResponse(diag[5], "Invalid data binding param in the signature : expected 'naMe', but found 'name'", 63);
+        assertResponse(diag[6], INVALID_RESOURCE_PARAMETERS, 65);
+        assertResponse(diag[7], INVALID_RESOURCE_PARAMETERS, 73);
```

Comment:
@m36dot aren't the old positions correct?
In L34 we have the error in details,
```
private static final String INVALID_RESOURCE_PARAMETERS =
            "invalid resource parameter(s): cannot specify > 2 parameters without specifying path config and/or body " +
                    "config in the resource annotation";
```
So I feel position should be the position of annotation. Not function def start. 
I fixed this to get annotation positions.

Reply:
@SupunS what do you think about this?

Reply:
Hmm interesting case. Yea looks like the old positions are correct.
I think we should first check which diag-position is being used to log the error.
If they are logging an error related to the annotation, but uses the pos of the function, then may be we should fix it there.